src/Compiler/Microsoft.NET.Sdk.Razor.SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -226,15 +227,28 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                 .Where(pair => pair.Right)
                 .Select((pair, _) => pair.Left);
 
+            var isAddComponentParameterAvailable = context.MetadataReferencesProvider
+                .Where(r => r.Display is { } display && display.EndsWith("Microsoft.AspNetCore.Components.dll", StringComparison.Ordinal))
+                .Collect()
+                .Select((refs, _) =>
+                {
+                    var compilation = CSharpCompilation.Create("components", references: refs);
```

Comment:
We shouldn't need to create a compilation here, this is going to be a pretty expensive operation. We should be doing nearly the same thing that we do with tag helpers from references above.

Reply:
I'm on the fence on this one (I suggested the change). This is only going to run when a ref is added or removed which is not very often at all vs checking the refs every time to determine if we want to run when the compilation changes which happens a lot. I guess if we really care we should run the benchmarks and see what the diff is, although I lean towards how its done right now.

Reply:
Especially if we filter the refs to just `Microsoft.AspNetCore.Components.dll` then it'll barely ever run.

Reply:
I guess I'll call in @jaredpar for another opinion.

Reply:
Short term I think this is fine. 

Longer term believe this is at least the second time this has come up. Essentially the inability to get symbols from references in a source generator is pretty limiting. It forces generators into awkward compromises like this. The actual hosts in every case are creating and caching these symbols ... we just can't get to them from the generators themselves. 

Did we file an issue to track exposing these better to consumers?

Reply:
https://github.com/dotnet/roslyn/issues/68993

Reply:
> Short term I think this is fine.

Alright, I shall concede. At least it won't run very often.